# Finks_Martiny_2022_in_review_mBio

Plasmids are key mobile genetic elements in bacterial evolution and ecology as they allow for the rapid adaptation of bacteria under selective environmental changes. However, the genetic information associated with plasmids is usually considered separately from information about their environmental origin. To broadly understand what kinds of traits become mobilized by plasmids in different environments, we analyzed the properties and accessory traits of 9,725 unique plasmid sequences from a publicly available database with known bacterial hosts and isolation sources. Although most plasmid research focuses on resistance traits, such genes made up less than 1% of the genes carried by plasmids. Similar to traits encoded on the bacterial chromosome, plasmid accessory trait composition (including general COG functions, resistance genes, and carbon and nitrogen genes) varied across seven major environments (human, animal, wastewater, plant, soil, marine, and freshwater). Despite their potential for horizontal gene transfer, plasmid traits were strongly influenced by their hostâ€™s evolutionary history. However, the trait differences across environments of broad COG categories could not be entirely explained by the plasmid host phylum, suggesting that environmental selection may be acting on the plasmid traits themselves. Lastly, some plasmid traits and environments (e.g., resistance genes in human-related environments) were more often associated with mobilizable plasmids than others. Overall, these findings underscore the high diversity of encoded traits by plasmids and provide a baseline for understanding the potential of plasmids to serve as reservoirs of a wide variety of adaptive traits for microbial communities.

This repository contains an R markdown file for scripts and packages used in the analysis described in the manuscript titled, "Plasmids as reservoirs of environmentally-adaptive bacterial traits". Additionally, PRIMER-e files having the PERMANOVA and statistical model used to disentangle the contribution of environment and host-phylum on variation of plasmid trait content are also included. This paper is currently in review at the ASM Journal mBio, and scripts/files will be revised accordingly.
